File: csharp_body_deref/Gisketch.Aottg2UI.Runtime/Gisketch.Aottg2UI.Building/GisketchBuildContext.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Gisketch.Aottg2UI.Building.GisketchBuildContext
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Gisketch/Aottg2UI/Building/GisketchBuildContext.c
// Prior real C# source: none (type is newer than the last public source snapshot)
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using System;
using System.Runtime.InteropServices;
using Gisketch.Aottg2UI.Actions;
using Gisketch.Aottg2UI.Data;
using Gisketch.Aottg2UI.Localization;
using Gisketch.Aottg2UI.State;
using Gisketch.Aottg2UI.Styling;
using Il2CppDummyDll;

namespace Gisketch.Aottg2UI.Building;

[Token(Token = "0x2000088")]
public sealed class GisketchBuildContext
{
	[Token(Token = "0x40002D5")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x10")]
	public readonly GisketchManifestDefinition Manifest;

	[Token(Token = "0x40002D6")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x18")]
	public readonly GisketchTheme Theme;

	[Token(Token = "0x40002D7")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x20")]
	public readonly IGisketchLocalizer Localizer;

	[Token(Token = "0x40002D8")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x28")]
	public readonly IGisketchActionHandler Actions;

	[Token(Token = "0x40002D9")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x30")]
	public readonly GisketchUIState State;

	[Token(Token = "0x40002DA")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x38")]
	public readonly GisketchComponentRegistry Registry;

	[Token(Token = "0x40002DB")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x40")]
	public readonly Func<string, GisketchOverlayDefinition> DialogResolver;

	[Token(Token = "0x60003C8")]
	[Address(RVA = "0x3A73280", Offset = "0x3A73280", VA = "0x3A73280")]
	public GisketchBuildContext(GisketchManifestDefinition manifest, GisketchTheme theme, IGisketchLocalizer localizer, IGisketchActionHandler actions, GisketchUIState state, [Optional] GisketchComponentRegistry registry, [Optional] Func<string, GisketchOverlayDefinition> dialogResolver)
	{
	}
}

File: cpp/Gisketch/Aottg2UI/Building/GisketchBuildContext.c
// Type: Gisketch.Aottg2UI.Building.GisketchBuildContext
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Gisketch.Aottg2UI.Runtime/Gisketch.Aottg2UI.Building/GisketchBuildContext.cs
// Prior real C# source: none
// --------------------------------

// Gisketch.Aottg2UI.Building.GisketchBuildContext$$.ctor
// il2cpp: void Gisketch_Aottg2UI_Building_GisketchBuildContext___ctor (Gisketch_Aottg2UI_Building_GisketchBuildContext_o* __this, Gisketch_Aottg2UI_Data_GisketchManifestDefinition_o* manifest, Gisketch_Aottg2UI_Styling_GisketchTheme_o* theme, Gisketch_Aottg2UI_Localization_IGisketchLocalizer_o* localizer, Gisketch_Aottg2UI_Actions_IGisketchActionHandler_o* actions, Gisketch_Aottg2UI_State_GisketchUIState_o* state, Gisketch_Aottg2UI_Building_GisketchComponentRegistry_o* registry, System_Func_string__GisketchOverlayDefinition__o* dialogResolver, const MethodInfo* method);
// 0x3b73280

void Gisketch_Aottg2UI_Building_GisketchBuildContext___ctor
               (Gisketch_Aottg2UI_Building_GisketchBuildContext_o *__this,
               Gisketch_Aottg2UI_Data_GisketchManifestDefinition_o *manifest,
               Gisketch_Aottg2UI_Styling_GisketchTheme_o *theme,
               Gisketch_Aottg2UI_Localization_IGisketchLocalizer_o *localizer,
               Gisketch_Aottg2UI_Actions_IGisketchActionHandler_o *actions,
               Gisketch_Aottg2UI_State_GisketchUIState_o *state,
               Gisketch_Aottg2UI_Building_GisketchComponentRegistry_o *registry,
               System_Func_string__GisketchOverlayDefinition__o *dialogResolver,MethodInfo *method)

{
  if (g_data_057a9e0f == '\0') {
    il2cpp_runtime_helper_023445d0(&TypeInfo_GisketchComponentRegistry);
    il2cpp_runtime_helper_023445d0(&TypeInfo_GisketchTheme);
    g_data_057a9e0f = '\x01';
  }
  System_Object___ctor((Il2CppObject *)__this,(MethodInfo *)0x0);
  (__this->fields).Manifest = manifest;
  il2cpp_runtime_helper_022b4080(&__this->fields,manifest);
  if (theme == (Gisketch_Aottg2UI_Styling_GisketchTheme_o *)0x0) {
    if (*(int *)(TypeInfo_GisketchTheme + 0xe4) == 0) {
      il2cpp_runtime_helper_02337ed0();
    }
    theme = (Gisketch_Aottg2UI_Styling_GisketchTheme_o *)**(undefined8 **)(TypeInfo_GisketchTheme + 0xb8);
  }
  (__this->fields).Theme = theme;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Theme,theme);
  (__this->fields).Localizer = localizer;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Localizer,localizer);
  (__this->fields).Actions = actions;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Actions,actions);
  (__this->fields).State = state;
  il2cpp_runtime_helper_022b4080(&(__this->fields).State);
  if (registry == (Gisketch_Aottg2UI_Building_GisketchComponentRegistry_o *)0x0) {
    if (*(int *)(TypeInfo_GisketchComponentRegistry + 0xe4) == 0) {
      il2cpp_runtime_helper_02337ed0();
    }
    registry = (Gisketch_Aottg2UI_Building_GisketchComponentRegistry_o *)
               **(undefined8 **)(TypeInfo_GisketchComponentRegistry + 0xb8);
  }
  (__this->fields).Registry = registry;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Registry,registry);
  (__this->fields).DialogResolver = dialogResolver;
  il2cpp_runtime_helper_022b4080(&(__this->fields).DialogResolver);
  return;
}


